pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -43,18 +44,20 @@ type ACMEIssuer struct {
 
 	// PrivateKey is the name of a secret containing the private key for this
 	// user account.
+	// +kubebuilder:validation:Required
 	PrivateKey cmmeta.SecretKeySelector `json:"privateKeySecretRef"`
 
 	// Solvers is a list of challenge solvers that will be used to solve
 	// ACME challenges for the matching domains.
-	// +optional
+	// +kubebuilder:validation:MinItems=1
```

Comment:
This needs to retain the `optional` entry - we don't actually require at least one solver to be specified. We require at least one of `http01` or `dns01` to be specified *within* a `ACMEChallengeSolver` structure, but not that at least one solver is defined.